metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1076,10 +1073,10 @@ public void handleLeaderChange(LeaderAndEpoch newLeader) {
                         renounce();
                     }
                 } else if (newLeader.isLeader(nodeId)) {
-                    log.info("Becoming the active controller at epoch {}, committed offset {}, " +
-                        "committed epoch {}", newLeader.epoch(), lastCommittedOffset,
-                        lastCommittedEpoch);
-                    claim(newLeader.epoch());
+                    long newLastWriteOffset = endOffset - 1;
```

Comment:
After an election, is the endOffset report by raft both the last written and last committed offset?

Reply:
The endOffset comes directly from the log and is as described... the end offset (exclusive).

Thinking about it more, I don't think I need to assume that it's committed, so I won't.

But it is used to calculate the next offset that the active controller should try to write to.